### ISSUE TITLE
Management operations using TokenCredential failing

### ISSUE THREAD

**Issue by SumiranAgg** [Azure/azure-cosmos-dotnet-v3#2394]
>We are continuously addressing and improving the SDK, if possible, make sure the problem persist in the [latest SDK version](https://www.nuget.org/packages/Microsoft.Azure.Cosmos).

**Describe the bug**
I gave an AAD identity DatabaseAccountContributor role from azure portal and /readMetadata and /container/* permissions as per RBAC control for dataplane operations (Referred this doc https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-setup-rbac). When trying to create a container using cosmos client initialized with this identity I see below error:

"The given request [POST /dbs/MetadataDatabase/colls] cannot be authorized by AAD token in data plane."

Am I doing something wrong or we do not have support to perform management operations using client created using AAD credentials?

**To Reproduce**
Covered above

**Expected behavior**
All operations should be successful.

**Actual behavior**
Create container call failed.

**Environment summary**
SDK Version: 3.18.0-preview
OS Version (e.g. Windows, Linux, MacOSX)

**Additional context**
Add any other context about the problem here (for example, complete stack traces or logs).

**Comment by j82w** [comment]
Currently the .NET and Java SDK do not support control plane operation via Azure Active Directory. Those [management operations](https://docs.microsoft.com/en-us/azure/cosmos-db/role-based-access-control) can be done via the via [ARM Templates](https://docs.microsoft.com/azure/cosmos-db/manage-with-templates), [PowerShell](https://docs.microsoft.com/azure/cosmos-db/manage-with-powershell), [Azure CLI](https://docs.microsoft.com/azure/cosmos-db/manage-with-cli), REST, or [Azure Management Library](https://github.com/Azure-Samples/cosmos-management-net).

**Comment by SumiranAgg** [comment]
Is the feature development on roadmap? Is yes do we have an ETA on when can we expect the support.

**Comment by j82w** [comment]
The feature is currently not being prioritized, because the control plane operations can be done via [Azure Management Library](https://github.com/Azure-Samples/cosmos-management-net) which has a c# nuget.

**Comment by qweeah** [comment]
Also met this issue when trying to move from using Cosmos DB string to Managed Identity in our services.

@j82w I can see that both `Azure.ResourceManager.CosmosDB` and `Microsoft.Azure.Management.CosmosDB` are available on Nuget, but the former is not updated for over 9 months, is it expected to be replaced by the latter?

> The feature is currently not being prioritized, because the control plane operations can be done via [Azure Management Library](https://github.com/Azure-Samples/cosmos-management-net) which has a c# nuget.

**Comment by rjygraham** [comment]
To understand why this needs to be prioritized, someone on the Cosmos DB team should do the following:

1. Follow the [Quickstart: Build a .NET console app to manage Azure Cosmos DB SQL API resources](https://docs.microsoft.com/en-us/azure/cosmos-db/create-sql-api-dotnet)
2. Run the quickstart and observe successful execution
3. Modify the quickstart to use DefaultAzureCredential/managed identity for authentication & grant RBAC permissions on the Cosmos DB account
4. Run the quickstart and observe all the failures
5. Go through the pain of modifying the quickstart to work 100% with managed identities leveraging multiple SDKs and authentication mechanisms.

**Comment by bedding** [comment]
I ran into the same issue lately and I am surprised this is not prioritized. The Azure.Cosmos SDK provides implementation for managing database and container while the service is blocking the operation from SDK, which seems to be inconsistent.

**Comment by blinard** [comment]
I understand that there's a difference between data-plane and control-plane RBAC but having a core part of this SDK's functionality broken when using TokenCredential for authentication is a _really_ poor developer experience. 

To add insult to injury, none of our documentation calls this out and our error messaging for this issue is fairly obtuse.

Could we, at the very least, take a priority story to update our documentation to reflect this constraint when using TokenCredentials? Also, perhaps better yet, could we throw a custom/clear exception (or output an obvious debugger/console message if an exception is deemed too risky) to callout that resource creation (databases, containers, etc.) are not supported when using TokenCredentials? I would think that we could identify the type of credentials being used from the implementation logic of the impacted methods to facilitate such messaging.

TokenCredential support was added to this SDK back in May. It's now 6 months later. I can understand not prioritizing the issue initially but it seems like we should do _something_ about it now. Even if it's just adding appropriate messaging and updating our docs to reflect the constraint. Obviously it'd be better to actually fix the issue b/c expecting devs to integrate multiple nuget packages and roll their own logic to ensure containers are created is a pretty poor experience too (especially when that concept is already supported within this sdk).

**Comment by ViSilver** [comment]
Are there any updates on the matter?

**Comment by bhanuprakash-1** [comment]
> I understand that there's a difference between data-plane and control-plane RBAC but having a core part of this SDK's functionality broken when using TokenCredential for authentication is a _really_ poor developer experience.
> 
> To add insult to injury, none of our documentation calls this out and our error messaging for this issue is fairly obtuse.
> 
> Could we, at the very least, take a priority story to update our documentation to reflect this constraint when using TokenCredentials? Also, perhaps better yet, could we throw a custom/clear exception (or output an obvious debugger/console message if an exception is deemed too risky) to callout that resource creation (databases, containers, etc.) are not supported when using TokenCredentials? I would think that we could identify the type of credentials being used from the implementation logic of the impacted methods to facilitate such messaging.
> 
> TokenCredential support was added to this SDK back in May. It's now 6 months later. I can understand not prioritizing the issue initially but it seems like we should do _something_ about it now. Even if it's just adding appropriate messaging and updating our docs to reflect the constraint. Obviously it'd be better to actually fix the issue b/c expecting devs to integrate multiple nuget packages and roll their own logic to ensure containers are created is a pretty poor experience too (especially when that concept is already supported within this sdk).

Totally Accept. Its very bad developer experience. Had spent a lot of time only to find that control plane operations are not supported when used TokenCredential(but wierdly is supported with other auth mechanisms). Atleast the error thrown should be pointing it out. 

Any update on this or can someone point me to a Learn tutorial where we can do cosmos DB management operations using .net SDK using TokenCredential...it would be really helpful. I expect something like has a very good documentation and a quick start tutorial when Azure itself recomends to use ManagedIdentitles based auth.

**Comment by johncrim** [comment]
> Currently the .NET and Java SDK do not support control plane operation via Azure Active Directory. Those [management operations](https://docs.microsoft.com/en-us/azure/cosmos-db/role-based-access-control) can be done via the via [ARM Templates](https://docs.microsoft.com/azure/cosmos-db/manage-with-templates), [PowerShell](https://docs.microsoft.com/azure/cosmos-db/manage-with-powershell), [Azure CLI](https://docs.microsoft.com/azure/cosmos-db/manage-with-cli), REST, or [Azure Management Library](https://github.com/Azure-Samples/cosmos-management-net).

@j82w - thank you, this is helpful. However, it really should be highlighted in docs/examples instead of left for developers to figure out/find this issue in github.

**Comment by hansmbakker** [comment]
Agree, it's a super bad developer experience to have the `client.CreateDatabaseIfNotExistsAsync` method available but getting runtime errors when using a `TokenCredential`.

Yes, this is [called out in the documentation](https://learn.microsoft.com/en-us/azure/cosmos-db/how-to-setup-rbac#permission-model:~:text=You%20cannot%20use%20any%20Azure%20Cosmos%20DB%20data%20plane%20SDK%20to%20authenticate%20management%20operations%20with%20an%20Azure%20AD%20identity).
No, it is not intuitive at all.
